File: cpp/2442_Count_Number_of_Distinct_Integers_After_Reverse_Operations/2442_Count_Number_of_Distinct_Integers_After_Reverse_Operations.cpp
class Solution {
public:
    int countDistinctIntegers(vector<int>& nums) {
        set<int> numbers;
        int size = nums.size();
        int result = 0;
        for (int i = 0; i < size; i++) {
            int reverse = reverseNumber(nums[i]);
            if (numbers.count(nums[i]) < 1) {
                numbers.insert(nums[i]);
                result += 1;
            }
            if (numbers.count(reverse) < 1) {
                numbers.insert(reverse);
                result += 1;
            }
        }
        
        return result;
    }
    
    int reverseNumber(int number) {
        int reversed = 0;
        while(number > 0) {
            reversed = reversed*10 + number%10;
            number /= 10;
        }
        return reversed;
    }
};

File: go/2442_Count_Number_of_Distinct_Integers_After_Reverse_Operations/2442_Count_Number_of_Distinct_Integers_After_Reverse_Operations.go
func countDistinctIntegers(nums []int) int {
    m := make(map[int]int)
    result := 0
    for _, v := range nums {
        _, exists := m[v]
        if !exists {
            m[v] = 1
            result += 1
        }
        reverse := reverseNumber(v)
        _, exists = m[reverse]
        if !exists {
            m[reverse] = 1
            result += 1
        }
    }
    return result
}

func reverseNumber(number int) int {
    reversed := 0
    for number > 0 {
        reversed = reversed*10 + number%10
        number /= 10
    }
    return reversed
}
